src/snowflake/connector/config_manager.py

```diff
@@ -217,21 +240,27 @@ def __init__(
         file_path: Path | None = None,
         _slices: list[ConfigSlice] | None = None,
     ):
-        """Create a new ConfigManager.
+        """Creates a new ConfigManager.
 
         Args:
             name: Name of this ConfigManager.
-            file_path: File this parser should read values from. Can be omitted
-              for all child parsers.
+            file_path: File this manager should read values from. Can be omitted
+              for all child managers.
+            _slices: List of ConfigSlices to consider. A configuration file's slice is a
+              section that can optionally reside in a different file. Note that this
+              feature might get deprecated soon.
         """
         if _slices is None:
             _slices = list()
         self.name = name
         self.file_path = file_path
         self._slices = _slices
-        # Objects holding subparsers and options
+        # Objects holding sub-managers and options
         self._options: dict[str, ConfigOption] = dict()
-        self._sub_parsers: dict[str, ConfigManager] = dict()
+        self._sub_managers: dict[str, ConfigManager] = dict()
+        # Alias for the old name of _sub_managers in the first release, please use the
+        #  new name, as this might get deprecated in the future.
+        self._sub_parsers = self._sub_managers
```

Comment:
nit: I think the depreciation information can be explicitly conveyed to users by logging a warning

Reply:
We synced up about this offline, I have added deprecation warnings and tests for these!